src/pages/_app.js

```diff
@@ -1,43 +1,42 @@
 import { useState } from "react";
 import { useRouter } from "next/router";
-import transactionsData from "@/src/data/transactions.json";
+import transactions from "@/src/data/transactions.json";
+import { uid } from "uid";
 
 export default function App({ Component, pageProps }) {
-  const [transactions, setTransactions] = useState(transactionsData);
+  const [initialData, setInitialData] = useState(transactions);
```

Comment:
Personally, I find 'initialData' confusing as a name here, since you're adding new transactions to this array. Just a comment.